### PR TITLE
Shell history enhancement

### PR DESCRIPTION
**Pull request by Emily82**
- Defined macros to handle posiz index for inspecting history;
- Added a missing return 0 for shell thread;
- Commands are saved to history only if the last saved one is different;
- When inspecting history, the first command showed is the last submitted.

### CODE REVIEW

File: src/include/shell/shell.h [hunk 1]
```diff
@@ -26,6 +26,18 @@
 #define DEF_COM 18
 #define HST_LEN 10
 
+#define RESET_POSIZ posiz = HST_LEN - 1;
```
Comment (inuyasha82) [inline src/include/shell/shell.h]: I don't think is a good idea to use macro that are abbreviation of italian words, please review macros and where possible also variable name (i.e. posiz should be pos, RESET_POSIZ should be RESET_POS etc...)

Comment (Emily82) [inline src/include/shell/shell.h]: Ok ill work on it

**Comment by inuyasha82** [comment]
Couple of comments:
1. it's not very clear its behaviour. I mean for example from the shell:
a. press up - it shows the last command
b. press down - it show another command (maybe the first one), i think if i press up and down it should clean the command (at least the first time, but maybe i consder acceptable that if you press up, then down and down again it brings you to the first element of the history). 
2. If enter several commands, then: 
a. enter for example the command help (or any command of your choice)
b. press up and surf to the previous one
c. press enter (it means i'm using the last command) 
d. press up again (it shows me the last command executed, correct)
e. press enter 
f. press up again (the last command should be help, or anyway the command you tiped on step a), but another command shows up (and apparently it isn't either the command typed before the last one). 

And what happens if i enter an empty line (didn't checked)?
Can you check it please?

**Comment by Emily82** [comment]
With empty line the command isn't saved. The buffer is circular, with only 10 slots, we could increase the amount if you want.
This means that after 10 writes you start to store new commands from the beginning, that's because you lose very old information.

**Comment by inuyasha82** [comment]
Still not working:
Enter couple of commands. 
Then press arrow up and press enter (use last command entered) 
Next time you press enter you will se the command before the last one. 

then as i pointed in my previous comment: if you press up you should scroll up, but if you press down we should at least clear the command from the shell (or scroll down the history, if it is not a big hassle).

**Comment by Emily82** [comment]
To fix the issue it has been needed to remove the feature that didn't save duplicate command in the history stack. I hope this is acceptable for you.

**Pull request closed without merging**